044-Stable-identities-for-Kafka-Connect-worker-nodes.md

```diff
@@ -0,0 +1,176 @@
+    1) Create a new Deployment with 0 replicas
+    2) Scale-up the Deployment by increasing the replica count by 1 to create a new Pod
+    3) Wait for the new Pod to become ready
+    4) Remove one Pod from the StrimziPodSet to get it removed
+    5) ... repeat steps 2-4 until the Deployment has the desired number of Pods and the StrimziPodSet has no Pods
+    6) Delete the StrimziPodSet
+* If the user is using the `Recreate` deployment strategy, Strimzi will
+    1) Create a new Deployment with 0 replicas
+    2) Remove one Pod from the StrimziPodSet to get it removed
+    3) Scale-up the Deployment by increasing the replica count by 1 to create a new Pod
+    4) Wait for the new Pod to become ready
+    5) ... repeat steps 2-4 until the Deployment has the desired number of Pods and the StrimziPodSet has no Pods
+    6) Delete the StrimziPodSet
+
+Doing the things in the right oder is important because otherwise it might happen that there will be no space in the cluster to start the new Pod and the migration might get stuck.
+
+#### Downgrades to Strimzi versions before the `StableConnectIdentities` feature gate
+
+Automatic downgrades from Strimzi version with enabled `StableConnectIdentities` feature gate to a Strimzi version which does not support the `StableConnectIdentities` feature gate will not be supported.
+Manual downgrade will be still possible.
+It will require the user to manually delete the `StrimziPodSet` used to manage the Kafka Connect pods and (possibly) also delete the corresponding Kafka Connect Pods.
+Deleting the Kafka Connect Pods manually will be needed only when downgrading to a Strimzi version which does not support or use StrimziPodSets.
+
+## Impacted projects / components
+
+This proposal impacts Kafka Connect and Kafka Mirror Maker 2 clusters.
+Any other operands (including Kafka Mirror Maker 1) or other Strimzi projects are not impacted by this.
+
+## Rejected alternatives
+
+### Using StatefulSets instead of StrimziPodSets
+
+I considered using StatefulSets instead of StrimziPodSets.
+Some of the issues we had with StatefulSets when using the for the Kafka brokers do not apply to Kafka Connect today.
+For example, we do not need asymmetric configuration of the worker nodes since Kafka Connect doesn't support advanced scheduling and you thus cannot ensure that some tasks get scheduled only on some _special_ nodes.
+
+However, this idea was ultimately rejected as StrimziPodSets might give us more options in the future.
+For example:
+* If we in the future support stretched Kafka clusters across multiple Kubernetes clusters, it might make sense to support stretched clusters also for Connect.
+  While you cannot easily schedule the tasks across the Pods running on the different Kubernetes clusters, having the workers running in all of will provide improved availability since the tasks can be more easily rescheduled if one of the Kubernetes clusters goes offline.
```

Comment:
```suggestion
  While you cannot easily schedule the tasks across the Pods running on the different Kubernetes clusters, having the workers running in all of the clusters will provide improved availability since the tasks can be more easily rescheduled if one of the Kubernetes clusters goes offline.
```